src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2163,6 +2163,36 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                 return;
             }
 
+            if (reason == LambdaConversionResult.MismatchedParameterDefaultValue)
+            {
+                Debug.Assert(anonymousFunction.ParameterCount == delegateParameters.Length);
+
+                var lambdaSymbol = anonymousFunction.TemporaryLambdaSymbol;
+                Debug.Assert(lambdaSymbol is not null);
+
+                // The lambda symbol may have diagnostics from
+                // binding default parameters, so copy these to the current diagnostic bag
+                lambdaSymbol.GetDeclarationDiagnostics(diagnostics);
+
+                for (int i = 0; i < anonymousFunction.ParameterCount; i++)
+                {
+                    var lambdaParamDefaultVal = lambdaSymbol.Parameters[i].ExplicitDefaultConstantValue;
+
+                    if (lambdaParamDefaultVal is not null && !lambdaParamDefaultVal.IsBad)
+                    {
+                        var delegateParamDefaultVal = delegateParameters[i].ExplicitDefaultConstantValue;
+                        if ((!delegateParamDefaultVal?.IsBad ?? true) && lambdaParamDefaultVal != delegateParamDefaultVal)
+                        {
+                            var lambdaParameterLocation = anonymousFunction.ParameterLocation(i);
+
+                            // Parameter {0} has default value '{1}' in lambda and '{2}' in target delegate type.
+                            Error(diagnostics, ErrorCode.ERR_OptionalParamValueMismatch, lambdaParameterLocation, i + 1, lambdaParamDefaultVal, delegateParamDefaultVal ?? ((object)"<missing>"));
```

Comment:
>"<missing>" [](http://example.com/codeflow?start=180&length=11)

This should be a localized string. See `MessageID.IDS_SK_TYPE.Localize()` for instance.